enhancements/installer/openstack-supported-versions.md

```diff
@@ -0,0 +1,195 @@
+    - [User Stories](#user-stories)
+      - [Check the OpenStack version](#check-the-openstack-version)
+      - [Skip the version check](#skip-the-version-check)
+        - [Example Usage](#example-usage)
+    - [Test Plan](#test-plan)
+    - [Graduation Criteria](#graduation-criteria)
+      - [Dev Preview -> Tech Preview](#dev-preview---tech-preview)
+      - [Tech Preview -> GA](#tech-preview---ga)
+    - [Infrastructure Needs](#infrastructure-needs)
+    - [Drawbacks](#drawbacks)
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [x] Graduation criteria
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+OpenShift can only be installed on supported versions of OpenStack.
+While the [Support Matrix](https://access.redhat.com/articles/4679401) is
+well documented, the OpenShift installer doesn't check which version of
+OpenStack is running before attempting an operation (e.g. installation). 
+Running OpenShift over an unsupported version of OpenStack is problematic
+and can lead to unstable infrastructure. To avoid that, it is proposed
+to programmatically verify the version of OpenStack is supported
+otherwise stop and fail.
+
+## Motivation
+
+### Goals
+
+- Disallow any kind of OpenShift operation done through the installer to run
+  against a cloud where the OpenStack version is not supported.
+- Allow an administrator to ignore the OpenStack version check and force the
+  installer to proceed anyway. This can be useful if the OpenStack cloud
+  is partially upgraded to a major version or if there is a Support
+  Exception granted to run that deprecated version of OpenStack.
+
+### Non-Goals
+
+This enhancement does not intent do block any OpenShift operations done
+outside of the installer if the OpenStack version is not supported.
+
+## Proposal
+
+In order to implement this feature fully, the following changes must be made
+in the openshift/installer:
+
+- Update gophercloud vendoring to (at least) [b56bf1](https://github.com/gophercloud/gophercloud/pull/2037/commits/5a1daf082451587459058cc539c53f2ae1aaaef3)
+  so we can beneficiate from the apiversions package.
+- Create a constant that will list all the supported versions
+  of Compute API based on the upstream manual.
+- Add a new validation against the OpenStack cloud to make sure that
+  the Compute API version is supported.
+- Create a configuration flag `skipVersionCheck` that when set to
+   false will disable the OpenStack version check and proceed anyway.
+
+### Design Details
+
+The OpenStack Compute service (also known as Nova) is always deployed
+in OpenStack clouds. Also, it is one of the services that bumps the
+microversions so regularly that is it possible to figure out
+what version of OpenStack is running based on the maximum version of
+Nova API that is available for a cloud.
```

Comment:
Ack. There's a contractual given here, which is: Nova will not bump the microversion of a stable release. There's also an assumption that Nova will always bump the microversion between stable releases (i.e. add or change the public API). This is a reasonable assumption which is unlikely to ever be broken. If it were ever not true we would still have an opportunity to look for alternatives.

Reply:
I'll add that comment to the details if you don't mind.